### PR TITLE
MOB-1235 - Created WC requests service

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**
Pass all WC requests through new service.
Handle requests one by one.
Updated all WC1 and WC2 sign functions into async. 
Updated request timeout logic, make it work for both WC. 
Unified WC errors.
Separated WC UI handlers for confirmation and errors. 
Show loading indicator and request timeout when sign receive sign request from deep link.
Wait for app to be in active state before handling requests.
Created WalletConnectExternalWalletSigner to pass requests to external wallet.
Set timer for expected response from external wallet.
Fixed issue when loading indicator was not dismissed when handle transaction in external wallet.
Tests for WCRequests service.

Fixed issue when single domain always run confetti on appear.

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Services/WalletConnect/WalletConnectService/WCRequestHandler.swift [hunk 1]
```diff
@@ -10,8 +10,10 @@ import WalletConnectSwift
 import Web3
 import UIKit
 
+// TODO: - WC Remove
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/WalletConnect/WalletConnectService/WCRequestHandler.swift:13]: dead code?

**Review by rommex — CHANGES_REQUESTED** [review]
WalletConnectExternalWalletSigner is only for WC1? Let's discuss

**Review by rommex — APPROVED** [review]

File: unstoppable-ios-app/domains-manager-ios/Entities/Wallets/UDWallet+Signing.swift [hunk 2]
```diff
@@ -117,79 +117,44 @@ extension UDWallet {
         if  sessions.count > 0 {
             let response = try await appContext.walletConnectServiceV2.sendPersonalSign(sessions: sessions,
                                                                                         message: message,
-                                                                                        address: address) {
-                try await launchExternalWallet()
-            }
+                                                                                        address: address,
+                                                                                        in: self)
```
[the remark this answers is not in the document]
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Entities/Wallets/UDWallet+Signing.swift:121]: Updated name of the class to better reflect it's purpose

File: unstoppable-ios-app/domains-manager-ios/Modules/DomainsCollection/DomainsCollectionCarouselItemViewController/DomainsCollectionCarouselItemViewPresenter.swift [hunk 3]
```diff
@@ -124,17 +123,17 @@ extension DomainsCollectionCarouselItemViewPresenter: AppLaunchServiceListener {
...
         
         Task {
             await showDomainDataWithActions(animated: true)
         }
     }
     
-    func didDisconnect(from app: PushSubscriberInfo?) {
-        guard app?.domainName == self.domain.name else { return }
+    func didDisconnect(from app: UnifiedConnectAppInfo) {
+        guard app.domain.name == self.domain.name else { return }
```
[the remark this answers is not in the document]
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/DomainsCollection/DomainsCollectionCarouselItemViewController/DomainsCollectionCarouselItemViewPresenter.swift]: Nice catch 👍

File: unstoppable-ios-app/domains-manager-ios/Services/WalletConnect/WalletConnectService/WCRequestHandler.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Services/WalletConnect/WalletConnectService/WCRequestHandler.swift]: Removed

**Pull request merged**